### PR TITLE
Make TensorInfo class compatible with OV 2.0 API

### PR DESCRIPTION
**Pull request by atobiszei**
Idea is to have internal OVMS precision to make potentiall mappings and other backends support easier.
It seems that the shape in 2.0 will be what we used and denoted earlier as effective shape.

JIRA:CVS-71639

### CODE REVIEW

File: src/node_library_utils.cpp [hunk 1]
```diff
@@ -43,25 +43,20 @@ CustomNodeTensorPrecision toCustomNodeTensorPrecision(InferenceEngine::Precision
...
-    case CustomNodeTensorPrecision::FP16:
-        return InferenceEngine::Precision::FP16;
-    case CustomNodeTensorPrecision::I16:
-        return InferenceEngine::Precision::I16;
-    case CustomNodeTensorPrecision::U16:
-        return InferenceEngine::Precision::U16;
-    default:
-        return InferenceEngine::Precision::UNSPECIFIED;
+Precision toInferenceEnginePrecision(CustomNodeTensorPrecision precision) {
+    std::unordered_map<CustomNodeTensorPrecision, Precision> precisionMap{
```
Comment (dkalinowski) [inline src/node_library_utils.cpp:47]: should be static

File: src/precision.hpp [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#pragma once
+
+#include <map>  // TODO remove
+#include <unordered_map>
+
+#include <inference_engine.hpp>
```
Comment (dkalinowski) [inline src/precision.hpp:21]: missing include of new ov

File: src/precision.hpp [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    };
+    auto it = precisionMap.find(precision);
+    if (it == precisionMap.end()) {
+        return ov::element::Type_t::undefined;  // TODO other way?
+    }
+    return it->second;
+}
+inline static Precision ovElementTypeToOvmsPrecision(ov::element::Type_t type) {
+    static std::unordered_map<ov::element::Type_t, Precision> precisionMap{
+        {ov::element::Type_t::f64, Precision::FP64},
```
Comment (dkalinowski) [inline src/precision.hpp:106]: Type_t => Type

Comment (atobiszei) [reply src/precision.hpp:106]: there is no Type::f64. Type_t is enum while Type is wraparound class for that enum. There is converting constructor Type(Type_t). I am not sure at this point which we should use in the context of OVMS so I would refrain with potential swapping until the end of 2.0 conversion when we have full picture.

**Review by dkalinowski — APPROVED** [review]

**Pull request merged**